app/src/main/java/fr/free/nrw/commons/WelcomeActivity.java

```diff
@@ -28,6 +30,11 @@ public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_welcome);
 
+
+        Bundle bundle = getIntent().getExtras();
```

Comment:
I would prefer a more conservative approach and check for `getIntent` `null` as well. Ref: https://stackoverflow.com/questions/37856407/can-activity-getintent-ever-return-null